dracut/30ignition/ignition-fetch.service

```diff
@@ -5,17 +5,17 @@ ConditionPathExists=/etc/initrd-release
 DefaultDependencies=false
 Before=ignition-complete.target
 After=basic.target
+ConditionPathExists=/run/ignition/neednet
 
-# Run after ignition-setup has run because ignition-setup
-# may copy in new/different ignition configs for us to consume.
-After=ignition-setup-base.service
-After=ignition-setup-user.service
-
-# Network may be used to fetch userdata content.
-After=network.target
+Requires=ignition-fetch-offline.service
+After=ignition-fetch-offline.service
 
 [Service]
 Type=oneshot
 RemainAfterExit=yes
 EnvironmentFile=/run/ignition.env
+# The reason for this awkward bit here is that having
+# Wants=network-online.target will cause systemd to always start networking
+# even if ConditionPathExists fails.
+ExecStart=/usr/bin/systemctl start network-online.target
```

Comment:
This is least hacky way I found to do dynamic enabling; need to think some more about whether there's a cleaner way to do this...

Reply:
there is probably I cath here, but couldn't we conditionalize the invocation of that service to be OnFailure of a target, something like:

ignition-offline-fetch.target
```
[Unit]
Description=Ignition (offline fetch) target
AssertPathExists=!/run/ignition/neednet
OnFailure=ignition-fetch.service
´´´

Reply:
another idea maybe (depending on the requirements of `fetch-offline`) is to not run NetworkManager via systemd, but rather still via the initqueue and run `fetch-offline` `Before=dracut-initqueue.service`

Reply:
the "stamp file" would then be writing the equivalent of `rd.neednet=1` into a file in /etc/cmdline.d

Reply:
> This is least hacky way I found to do dynamic enabling; need to think some more about whether there's a cleaner way to do this...

We have decent contacts with the `systemd` team, could we reach out to them and ask for their advice?  Maybe it ends up being an RFE to `systemd`?

Reply:
There is this issue which is essentially what we're hitting here (also opened by somebody very well known to us): https://github.com/systemd/systemd/issues/985